### ISSUE TITLE
Would be nice to diagnose missmatched plant loop volume and operating flow rates. (CR #9355)

### ISSUE THREAD

**Issue by axelstudios** [NREL/EnergyPlus#4126]
###### Added on 2013-10-24 16:14 by Brent Griffith
--
#### Description
User had a difficult to identify problem where the autocalculated plant loop volume was very large compared to the flow rate.  Supply fluid reset strategies did not work because the loop capacitance was so large that the rest neve makes it to the demand side.  Other users have found that for domestic hot water systems, the autocalculation produces such a large capacitance that the mains water temperature model does not work well enough.  

Suggestion is to find a way to diagnose missmatched sizing of plant loop volume compared to operating flow rates.  Approaches to investigate include:
1. monitor temperature differences between supply outlet and demand inlet and detect if they persist over long periods of time.
2. monitor the maximum operating flow rate and compare that to the magnitude of impact on the capacitance model.  Some criteria to compare Mdot*c*DeltaT flow term to McDT/Dt term.

--
External Ref: Ticket 5556
Last build tested: `13.10.18 V8.1.0.008`

**Comment by rglazer2** [comment]
@mjwitte Any more advice on exactly what needs to be done for this fix, and what would be the best way to accomplish it?

**Comment by mjwitte** [comment]
@EnergyArchmage Any suggestions here?  Or would you rather take this one?

**Comment by EnergyArchmage** [comment]
Hi @rglazer2, here is my $0.02

note that issue #5516 is also related to this and has some discussion
1. change up how the autocalculate sizes plant volume, if they come out smaller this problem gets less. #4188 
2.  add some output variables so time series reporting can more easily diagnose. I suggest adding some calculations and data storage to the routine called UpdateHalfLoopInletTemp. 

I would add two new variables to the plant loop/loop side data structure, along side of LoopSideInlet_TankTemp.  Variables called something like LoopSideInlet_MdotCpDeltaT and LoopSideInlet_McpDTdt, with output names like (for both loop sides)
Plant Demand Side Lumped Capacitance Heat Transport Rate [W]
Plant Demand Side Lumped Capacitance Heat Storage Rate [W]
Plant Supply Side Lumped Capacitance Heat Transport Rate [W]
Plant Supply Side Lumped Capacitance Heat Storage Rate [W]

The calculations should be easy there with the newly calculated TankFinalTemp and all the terms like LastTankOutletTemp available, something like this at the end of the routine. 

```
    PlantLoop( LoopNum ).LoopSide( TankOutletLoopSide ).LoopSideInlet_MdotCpDeltaT = (TankInletTemp-TankFinalTemp)* Cp*MassFlowRate;
    PlantLoop( LoopNum ).LoopSide( TankOutletLoopSide ).LoopSideInlet_McpDTdt = ThisTankMass * Cp * (TankFinalTemp -  LastTankOutletTemp);
```
1.  Now the tricky thing would be use these new values to throw a warning.  A nicely running plant loop would typically have LoopSideInlet_MdotCpDeltaT > LoopSideInlet_McpDTdt most of the time.  Times when LoopSideInlet_MdotCpDeltaT < LoopSideInlet_McpDTdt are expected to happen though.   the problem is when LoopSideInlet_MdotCpDeltaT < LoopSideInlet_McpDTdt is happeing the majority of the time during the simulation.  Need a  good idea for annual aggregation and detecting for a warning at the end. 

I could take these on if you want, or at least help to review.

**Comment by rglazer2** [comment]
Thanks for the suggestions @EnergyArchmage. I added the those variables you suggested, and their calculations in the UpdateHalfLoopInletTemp function. 

I was thinking that after LoopSideInlet_MdotCpDeltaT and LoopSideInlet_McpDTdt are calculated I could check if LoopSideInlet_MdotCpDeltaT < LoopSideInlet_McpDTdt, and then add a new variable which would be incremented each time that is the case. Then at the end of the simulation I could check if that occurred more than half(?) of the time, and if it does report a warning. I was hoping you could advise on where that check/warning should occur? Is there a way to check that it's in the final iteration? Does this seem like a good approach, or would another way be better?

**Comment by EnergyArchmage** [comment]
Yeah, I was think another output variable with units of hours.  It would be a sum-type variable that holds the current fractional hours value for times when LoopSideInlet_MdotCpDeltaT < LoopSideInlet_McpDTdt.  I am not sure what to call them, but here is a suggestion

Plant Supply Side Lumped Capacitance Excessive Storage Time [ hr ]
Plant Demand Side Lumped Capacitance Excessive Storage Time [ hr ]

I would think the call for the check/warning would go in somewhere around line 640 in SimulationManager.cc.

**Comment by EnergyArchmage** [comment]
Oh and you may need to use the same sort of code as is used in the annual tabular reporting to aggregate values over the simulation.  It may make sense to add this to some predefined annual summary table, maybe @JasonGlazer  has an idea.

**Comment by Myoldmopar** [comment]
Addressed via #5791
